provisioning/baremetal_pod.go

```diff
@@ -468,10 +475,53 @@ func NewMetal3Deployment(targetNamespace string, images *Images, config *metal3i
 	}
 }
 
-func GetExistingMetal3Deployment(client appsclientv1.DeploymentsGetter, targetNamespace string) (bool, error) {
-	existing, err := client.Deployments(targetNamespace).Get(context.Background(), baremetalDeploymentName, metav1.GetOptions{})
+func annotateExistingMetal3Deployment(existing *appsv1.Deployment) {
+	curAnnotations := existing.GetAnnotations()
+	if _, ok := curAnnotations[cboOwnedAnnotation]; ok {
+		// metal3 deployment already owned by CBO
+		return
+	}
+	cboAnnotations := map[string]string{}
+	cboAnnotations[cboOwnedAnnotation] = ""
+	modified := resourcemerge.BoolPtr(true)
+	resourcemerge.MergeMap(modified, &cboAnnotations, existing.GetAnnotations())
+	existing.SetAnnotations(cboAnnotations)
+	return
+}
+
+func ApplyNewOrUpdatedDeployment(client appsclientv1.DeploymentsGetter, targetNamespace string, metal3Deployment *appsv1.Deployment, generations []osoperatorv1.GenerationStatus, componentName string) error {
+
+	// Check if there is an existing metal3 deployment
+	existing, err := client.Deployments(targetNamespace).Get(context.Background(), BaremetalDeploymentName, metav1.GetOptions{})
+
 	if existing != nil && err == nil {
```

Comment:
can we make the function not have so much duplicated code?

Reply:
Done.